crates/sui-core/src/authority_active.rs

```diff
@@ -233,9 +234,28 @@ where
     pub async fn spawn_gossip_process(self, degree: usize) -> JoinHandle<()> {
         let active = Arc::new(self);
 
-        let gossip_locals = active;
+        // Number of tasks at most "degree" and no more than committee - 1
+        // (validators do not follow themselves for gossip)
+        let committee = active.state.committee.load().deref().clone();
+        let target_num_tasks = usize::min(committee.voting_rights.len() - 1, degree);
+
+        tokio::task::spawn(async move {
+            gossip_process(&active, target_num_tasks).await;
+        })
+    }
+
+    pub async fn spawn_node_sync_process(self) -> JoinHandle<()> {
+        let active = Arc::new(self);
+        let committee = active.state.committee.load().deref().clone();
+        // nodes follow all validators to ensure they can eventually determine
+        // finality of certs. We need to follow 2f+1 _honest_ validators to
+        // eventually find finality, therefore we must follow all validators.
+        //
+        // TODO: does this still apply when weighing by stake?
```

Comment:
Is this TODO still relevant? I think it does apply right?

Reply:
Probably? I'm so used to ignoring stake when thinking about this stuff that I'm not confident reasoning about stake. But yes, I think so, and I'll remove the TODO.